printThis.js

```diff
@@ -41,295 +41,317 @@
+          // check if the iframe was created with the ugly hack
+          // and perform another ugly hack out of neccessity
+          window.frames["printIframe"].focus();
+          $head.append("<script>  window.print(); </s" + "cript>");
+        } else {
+          // proper method
+          if (document.queryCommandSupported("print")) {
+            $iframe[0].contentWindow.document.execCommand("print", false, null);
+          } else {
+            $iframe[0].contentWindow.focus();
+            $iframe[0].contentWindow.print();
+          }
         }
 
-        // $iframe.ready() and $iframe.load were inconsistent between browsers
-        setTimeout(function() {
-
-            // Add doctype to fix the style difference between printing and render
-            function setDocType($iframe, doctype){
-                var win, doc;
-                win = $iframe.get(0);
-                win = win.contentWindow || win.contentDocument || win;
-                doc = win.document || win.contentDocument || win;
-                doc.open();
-                doc.write(doctype);
-                doc.close();
-            }
-
-            if (opt.doctypeString){
-                setDocType($iframe, opt.doctypeString);
-            }
-
-            var $doc = $iframe.contents(),
-                $head = $doc.find("head"),
-                $body = $doc.find("body"),
-                $base = $('base'),
-                baseURL;
-
-            // add base tag to ensure elements use the parent domain
-            if (opt.base === true && $base.length > 0) {
-                // take the base tag from the original page
-                baseURL = $base.attr('href');
-            } else if (typeof opt.base === 'string') {
-                // An exact base string is provided
-                baseURL = opt.base;
-            } else {
-                // Use the page URL as the base
-                baseURL = document.location.protocol + '//' + document.location.host;
-            }
-
-            $head.append('<base href="' + baseURL + '">');
-
-            // import page stylesheets
-            if (opt.importCSS) $("link[rel=stylesheet]").each(function() {
-                var href = $(this).attr("href");
-                if (href) {
-                    var media = $(this).attr("media") || "all";
-                    $head.append("<link type='text/css' rel='stylesheet' href='" + href + "' media='" + media + "'>");
-                }
-            });
+        // remove iframe after print
+        if (!opt.debug && !opt.keepIframe) {
+          setTimeout(function() {
+            $iframe.remove();
+          }, 1000);
+        }
 
-            // import style tags
-            if (opt.importStyle) $("style").each(function() {
-                $head.append(this.outerHTML);
-            });
+        // after print callback
+        if (typeof opt.afterPrint === "function") {
+          if (opt.keepIframe) {
+            $iframe.remove();
```

Comment:
This would fire even sooner, because `afterprint` does not wait; it depends on the browser interrupting execution.